src/Psl/Locale/Locale.php

```diff
@@ -0,0 +1,896 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Locale;
+
+use Locale as NativeLocale;
+
+/**
+ * Represents a locale identifier.
+ */
+enum Locale: string
```

Comment:
The cases for this enum were mostly generated using a script, we could add more in the future.

Reply:
Would it make sense to include said script and check from time to time that new locales are added automatically?

Reply:
This is what I used actually, but had to go over the result manually and update some names as they contained non-ASCII characters.

I think it would be nice to have a nice script for this in the future, but I don't see the value on having this in the repo.

```php
foreach (\ResourceBundle::getLocales('') as $locale) {
    $name = \Locale::getDisplayName($locale, 'en');
    $name = \Psl\Str\replace_every($name, [
        '(' => '',
        ')' => '',
        '[' => '',
        ']' => '',
        ',' => '',
        '.' => '',
        '&' => '',
        '-' => '',
        ' ' => '',
    ]);

    echo 'case ' . $name . ' = "' . $locale . '";' . PHP_EOL;
}
```